sdks/java/harness/src/main/java/org/apache/beam/fn/harness/FnApiDoFnRunner.java

```diff
@@ -384,6 +384,7 @@
     this.windowingStrategy = windowingStrategy;
     this.doFnSignature = DoFnSignatures.signatureForDoFn(doFn);
     this.doFnInvoker = DoFnInvokers.invokerFor(doFn);
+    this.doFnInvoker.invokeSetup();
```

Comment:
Do we have precedent for invoking setup in the constructor like this? Is this the correct place to do it? Note that if this prevents us from serializing and sending off the DoFn runner while expecting setup to have been completed wherever it is reinstantiated.

Reply:
I wasn't sure how the DoFnRunner was used. That makes sense.